[unifi] New site, wlan, wiredClient, and poePort. Discovery support (#11959)
openhab/openhab-addons: bundles/org.openhab.binding.unifi/README.md

_Note: All channels with the Write permission require administrator credentials as defined in the controller._

### `wiredClient`

The `wiredClient` information that is retrieved is available as these channels:

| Channel ID | Item Type | Description                                                          | Permissions |
|------------|-----------|----------------------------------------------------------------------|-------------|
| online     | Switch    | Online status of the client                                          | Read        |
| site       | String    | Site name (from the controller web UI) the client is associated with | Read        |
| macAddress | String    | MAC address of the client                                            | Read        |
| ipAddress  | String    | IP address of the client                                             | Read        |
| uptime     | Number    | Uptime of the client (in seconds)                                    | Read        |
| lastSeen   | DateTime  | Date and Time the client was last seen                               | Read        |

> | experience | Number    | Experience of the client                                             | Read        |

Reviewer: Would it be possible to document the meaning of this number?

Reviewer: @Hilbrand - any update here? Last open comment.

| blocked    | Switch    | Blocked status of the client                                         | Read, Write |

##### `blocked`

The `blocked` channel allows you to block / unblock a client via the controller.

##### `reconnect`

The `reconnect` channel allows you to force a client to reconnect.
Sending `ON` to this channel will trigger a reconnect via the controller.

### `poePort`

The `poePort` information that is retrieved is available as these channels:

| Channel ID | Item Type                | Description                                        | Permissions |
|------------|--------------------------|----------------------------------------------------|-------------|
| online     | Switch                   | Online status of the port                          | Read        |
| mode       | Selection                | Select the PoE mode: off, auto, 24v or passthrough | Read, Write |

> | enable     | Switch                   | Enable Power Over Ethernet                         | Read, Write |

Reviewer: Also here: "Power **o**ver Ethernet".

##### `considerHome`

The `considerHome` parameter allows you to control how quickly the binding marks a client as away.
For example, using the default of `180` (seconds), the binding will report a client away as soon as `lastSeen` + `180` (seconds) < `now`.

### `poePort`

The following table describes the `poePort` configuration parameters:

| Parameter   | Description                                                 | Config   |
| ------------|-------------------------------------------------------------|----------|

> | portIdx     | The port number as reported by the switch                   | Required |

Reviewer: Perhaps portNumber as name would be more straight-forward? Can it be mentioned in the description if they are numbered from 0 or 1?

| macAddress  | The MAC address of the switch device the port is part of    | Required |

## Channels

### `site`

The `site` information that is retrieved is available as these channels:

| Channel ID      | Item Type | Description                          | Permissions |
|-----------------|-----------|--------------------------------------|-------------|
| totalClients    | Number    | Total number of clients connected    | Read        |
| wirelessClients | Number    | Number of wireless clients connected | Read        |
| wiredClients    | Number    | Number of wired clients connected    | Read        |
| guestClients    | Number    | Number of guest clients connected    | Read        |

### `wlan`

The `wlan` information that is retrieved is available as these channels:

| Channel ID      | Item Type | Description                                                                    | Permissions |
|-----------------|-----------|--------------------------------------------------------------------------------|-------------|

> | enable          | Switch    | Enable status of the wLAN                                                      | Read, Write |

Reviewer: ```suggestion
| enable          | Switch    | Enable status of the WLAN                                                      | Read, Write |
```

| wirelessClients | Number    | Number of wireless clients connected                                           | Read        |
| guestClients    | Number    | Number of guest clients connected                                              | Read        |
| essid           | String    | Wireless Network (ESSID)                                                       | Read        |
| site            | String    | UniFi Site the client is associated with                                       | Read        |
| security        | String    | Security protocol of the Wi-Fi network                                         | Read        |
| wlanBand        | String    | Wireless LAN band of the Wi-Fi network                                         | Read        |
| wpaEnc          | String    | WPA Encoding of the Wi-Fi network                                              | Read        |
| wpaMode         | String    | WPA Mode of the Wi-Fi network                                                  | Read        |

> | xPassphrase     | String    | Passphrase of the Wi-Fi network                                                | Read        |

Reviewer: What does the x in xPassphrase denote?

| qrcodeEncoding  | String    | MECARD like encoding to generate a QRCode for easy access to the Wi-Fi network | Read        |

::: warning Attention
If you link an item to the `xPrassphrase` or `qrcodeEncoding` channel your Wi-Fi password will be  exposed in openHAB.
The password will also be visible in openHAB event log.
:::

> The `qrcodeEncoding` channel can be used to easily create a QRCode to access, for example, a guest network.

Reviewer: ```suggestion
The `qrcodeEncoding` channel can be used to easily create a QR Code to access, for example, a guest network.
```

It contains a MECARD like representation of the access.

> This is the notation used in QRCodes that can be scanned by mobile phones.

Reviewer: ```suggestion
This is the notation used in QR Codes that can be scanned by mobile phones.
```

| guestClients    | Number    | Number of guest clients connected                                               | Read        |
| essid           | String    | Wireless Network (ESSID)                                                        | Read        |
| site            | String    | UniFi Site the client is associated with                                        | Read        |
| security        | String    | Security protocol of the Wi-Fi network                                          | Read        |
| wlanBand        | String    | Wireless LAN band of the Wi-Fi network                                          | Read        |
| wpaEnc          | String    | WPA Encoding of the Wi-Fi network                                               | Read        |
| wpaMode         | String    | WPA Mode of the Wi-Fi network                                                   | Read        |
| passphrase      | String    | Passphrase of the Wi-Fi network                                                 | Read        |
| qrcodeEncoding  | String    | MECARD like encoding to generate a QR Code for easy access to the Wi-Fi network | Read        |

::: warning Attention

> If you link an item to the `prassphrase` or `qrcodeEncoding` channel your Wi-Fi password will be  exposed in openHAB.

Reviewer: Still one **r** too much:
```suggestion
If you link an item to the `passphrase` or `qrcodeEncoding` channel your Wi-Fi password will be exposed in openHAB.
```

:-)

> If you link an item to the `xPrassphrase` or `qrcodeEncoding` channel your Wi-Fi password will be  exposed in openHAB.

Reviewer: ```suggestion
If you link an item to the `passphrase` or `qrcodeEncoding` channel your Wi-Fi password will be exposed in openHAB.
```

Reviewer: Still one **r** too much in the word "p**r**assphrase".

| ipAddress  | String               | IP address of the client                                             | Read        |
| guest      | Switch               | On if this is a guest client                                         | Read        |
| ap         | String               | Access point (AP) the client is connected to                         | Read        |
| essid      | String               | Network name (ESSID) the client is connected to                      | Read        |
| rssi       | Number               | Received signal strength indicator (RSSI) of the client              | Read        |
| uptime     | Number               | Uptime of the client (in seconds)                                    | Read        |
| lastSeen   | DateTime             | Date and Time the client was last seen                               | Read        |
| experience | Number:Dimensionless | Overall health indication of the client (in percentage)              | Read        |
| blocked    | Switch               | Blocked status of the client                                         | Read, Write |

> | cmd        | String               | Command channel: `reconnect` to force the client to be reconnect     | Write       |

Reviewer: ```suggestion
| cmd        | String               | Command channel: `reconnect` to force the client to reconnect        | Write       |
```

(or: "to be reconnected")

> | reconnect  | Switch               | Force the client to be reconnect                                     | Write       |

Reviewer: ```suggestion
| reconnect  | Switch               | Force the client to reconnect                                        | Write       |
```

(or: "to be reconnected")

_Note: All channels with the Write permission require administrator credentials as defined in the controller._

### `wiredClient`

The `wiredClient` information that is retrieved is available as these channels:

| Channel ID | Item Type            | Description                                                          | Permissions |
|------------|----------------------|----------------------------------------------------------------------|-------------|
| online     | Switch               | Online status of the client                                          | Read        |
| site       | String               | Site name (from the controller web UI) the client is associated with | Read        |
| macAddress | String               | MAC address of the client                                            | Read        |
| ipAddress  | String               | IP address of the client                                             | Read        |
| uptime     | Number               | Uptime of the client (in seconds)                                    | Read        |
| lastSeen   | DateTime             | Date and Time the client was last seen                               | Read        |

> | experience | Number:Dimensionless | Experience of the client                                             | Read        |

Reviewer: ```suggestion
| experience | Number:Dimensionless | Experience of the client (in percentage)                             | Read        |
```

| essid          | String    | Wireless Network (ESSID)                                                       | Read        |
| site           | String    | UniFi Site the client is associated with                                       | Read        |
| security       | String    | Security protocol of the Wi-Fi network                                         | Read        |
| wlanBand       | String    | Wireless LAN band of the Wi-Fi network                                         | Read        |
| wpaEnc         | String    | WPA Encoding of the Wi-Fi network                                              | Read        |
| wpaMode        | String    | WPA Mode of the Wi-Fi network                                                  | Read        |
| xPassphrase    | String    | Passphrase of the Wi-Fi network                                                | Read        |
| qrcodeEncoding | String    | MECARD like encoding to generate a QRCode for easy access to the Wi-Fi network | Read        |

::: warning Attention
If you link an item to the `xPrassphrase` or `qrcodeEncoding` channel your Wi-Fi password will be  exposed in openHAB.

> The passwords is also visible in openHAB event log.

Reviewer: singular: password

| ipAddress  | String    | IP address of the client                                             | Read        |
| uptime     | Number    | Uptime of the wireless client (in seconds)                           | Read        |
| lastSeen   | DateTime  | Date and Time the wireless client was last seen                      | Read        |
| blocked    | Switch    | Blocked status of the client                                         | Read, Write |

##### `blocked`

The `blocked` channel allows you to block / unblock a client via the controller.

##### `reconnect`

The `reconnect` channel allows you to force a client to reconnect.
Sending `ON` to this channel will trigger a reconnect via the controller.

### `poePort`

The `poePort` information that is retrieved is available as these channels:

| Channel ID | Item Type                | Description                                        | Permissions |
|------------|--------------------------|----------------------------------------------------|-------------|
| online     | Switch                   | Online status of the port                          | Read        |

> | mode       | Selection                | Select the POE mode: off, auto, 24v or passthrough | Read, Write |

Reviewer: Usually spelled P**o**E.

| enable     | Switch                   | Enable Power Over Ethernet                         | Read, Write |
| power      | Number:Power             | Power consumption of the port in Watt              | Read        |
| voltage    | Number:ElectricPotential | Voltage of the port in Volt                        | Read        |
| current    | Number:ElectricCurrent   | Current used by the port in mA                     | Read        |

> The `enable` switch channel has a configuration parameter `mode` which is the value used to switch POE on when the channel is switched to ON.

Reviewer: Usually spelled P**o**E.
